### ISSUE TITLE
[cli] add `deploy --build` flag to pre-build islands

### ISSUE THREAD

**Issue by miguelrk** [netzo/netzo#33]
**TLDR:** add `netzo deploy --build` flag to [pre-build islands](https://github.com/denoland/fresh/issues/1062) AOT (ahead-of-time) rather than default JIT (just-in-time) which is apparently now [discouraged and will be removed soon](https://discord.com/channels/684898665143206084/991511118524715139/1161618652366393345).

### Context

From [Fresh 1.4 blogpost](https://deno.com/blog/fresh-1.4): 
> So far, Fresh has always compiled assets on the fly. This has served us well so far, as it enables lightning fast deployments with no build step, but we realised just-in-time (=JIT) rendering with large islands was noticeably slower. We arrived at a pre-compile solution that results in assets being served **~45-60x** faster for a cold start of a serverless function, with minimal impact on deployment times. The savings depend on the size of the island, but even for small ones the improvements are very visible.
> For example, a search box island in the fresh docs revives nearly instantaneously with AOT compiled assets whereas it took 4.28s with JIT compilation.

### Context

DD supports 4 deployment options and we currently implement the following (see [relevant issue)](https://github.com/netzo/netzo/issues/18):

- [x] **1. Deploy within Netzo:** enable quick edits and simple deployments (alpha: not the best DX in editor)
- [x] **2. Deploy with `netzo/cli`:** enables local development and manual deployment via `netzo deploy`
- [ ] **3. Deploy with Github integration:** enables automatic deployment and source control via GitHub
  - [ ] **3a. Automatic:** link a repo+branch in app, set up webhook in github, and let api listen to new commits on push
  - [ ] **3b. Github Action:** ability to customize deployment with additional steps in workflow (advanced)
  
[Fresh 1.4](https://deno.com/blog/fresh-1.4) introduced support for pre-building islands but it currently only works via **3b. Github Action** integration of DD. Marvin mentioned that the DD team is experimenting on adding automatic builds for the **3a. Automatic:** integration. While **3b** must spin-up a GitHub Action for the build (takes ~minutes), builds for **3a** could be implemented in an isolate, which would make them much, much faster (maybe ~seconds). This will reduce the greatest downside of having build-step, that it takes time.

### Solution

In any case, **an option to build should be seamlessly integrated into the deployment workflow for best DX.** This can be done directly in the `deploy` subcommand of the Netzo CLI, and it should be implemented as a regular script (e.g. user can execute `deno task build` and `deno task preview` locally), rather than baked-into the Netzo platform/infrastructure. Alternatively, we could also implement this at the Netzo API, so that not only deployments from the CLI, but also from the studio can be built.

1. 🟢 **Build at CLI:** can be easily implemented via deno scripts, user can manually build their projects and preview them locally and have full control over the build pipeline. This also integrates better into CI/CD workflows like GitHub Actions.
2. 🟡  **Build at API:** Could allow building deployments made from studio, but other than that, not many more benefits

### Relevant Links

- [Discussion about a fresh build step](https://discord.com/channels/684898665143206084/991511118524715139/1143179738911875092) (discussion starts here and goes on for several messages)
- https://github.com/denoland/fresh/issues/1700
